server/src/main/java/io/spine/server/EnvSetting.java

```diff
@@ -193,13 +197,31 @@ void reset() {
     void use(V value, Class<? extends EnvironmentType> type) {
         checkNotNull(value);
         checkNotNull(type);
-        this.environmentValues.put(type, value);
+        this.environmentValues.put(type, new Value<>(value));
+    }
+
+    /**
+     * Sets the value lazily provided via the passed {@code Supplier}
+     * for the specified environment type.
+     *
+     * <p>The supplier will not be invoked unless someone requests the value under
+     * the matching environment.
+     *
+     * @param value
+     *         supplier of the value to assign to one of environments
+     * @param type
+     *         the type of the environment
+     */
+    void lazyUse(Supplier<V> value, Class<? extends EnvironmentType> type) {
```

Comment:
Let's just call it `use`. The compiler will allow it because the args are different and, I think, it reads better as well.

Also, let's put the functional interface param last. This way, in Kotlin, we can use the DSL-like syntax.